File: 04.Simulation模拟/leetcode_2452_twoEditWords.cpp
#include <iostream>
#include <vector>
#include <string>
using namespace std;

class Solution {
public:
    vector<string> twoEditWords(vector<string>& queries, vector<string>& dictionary) {
        vector<string> res;
        for (string s : queries){
            bool flag = false;
            for (string d : dictionary){
                if (s.size() != d.size()){
                    continue;
                }
                int cnt = 0;
                for (int i=0; i<s.size(); i++){
                    if (s[i] != d[i]){
                        cnt++;
                    }
                }
                if (cnt <= 2){
                    flag = true;
                    break;
                }
            }
            if (flag){
                res.push_back(s);
            }
        }
        return res;
    }
};

int main(){
    Solution s;
    vector<string> queries = {"word","note","ants","wood"}, dictionary = {"wood","joke","moat"};
    vector<string> res = s.twoEditWords(queries, dictionary);
    for (auto& s : res){
        cout << s << " ";
    }
    cout << endl;
}

File: 04.Simulation模拟/leetcode_2452_twoEditWords.py
from typing import List

class Solution:
    def twoEditWords(self, queries: List[str], dictionary: List[str]) -> List[str]:
        res = []
        for s in queries:
            flag = False
            for d in dictionary:
                if len(s) != len(d):
                    continue
                if sum(1 for i in range(len(s)) if s[i] != d[i]) <= 2:
                    flag = True
                    break
            if flag:
                res.append(s)
        return res

if __name__ == "__main__":
    s = Solution()
    queries = ["word","note","ants","wood"]
    dictionary = ["wood","joke","moat"]
    print(s.twoEditWords(queries, dictionary))
